src/storage/mod.rs

```diff
@@ -994,19 +1074,57 @@ impl Storage {
         Ok(())
     }
 
+    pub fn async_raw_batch_delete(
+        &self,
+        ctx: Context,
+        keys: Vec<Vec<u8>>,
+        callback: Callback<()>,
+    ) -> Result<()> {
+        for key in &keys {
+            if key.len() > self.max_key_size {
+                callback(Err(Error::KeyTooLarge(key.len(), self.max_key_size)));
+                return Ok(());
+            }
+        }
+        let requests = keys.into_iter()
+            .map(|k| Modify::Delete(CF_DEFAULT, Key::from_encoded(k)))
+            .collect();
+        self.engine
+            .async_write(&ctx, requests, box |(_, res): (_, engine::Result<_>)| {
+                callback(res.map_err(Error::from))
+            })?;
+        RAWKV_COMMAND_COUNTER_VEC
+            .with_label_values(&["raw_batch_delete"])
+            .inc();
+        Ok(())
+    }
+
     fn raw_scan(
-        snapshot: Box<Snapshot>,
+        snapshot: &Snapshot,
         start_key: &Key,
+        end_key: Option<Key>,
         limit: usize,
         stats: &mut Statistics,
+        key_only: bool,
     ) -> engine::Result<Vec<Result<KvPair>>> {
-        let mut cursor = snapshot.iter(IterOption::default(), ScanMode::Forward)?;
+        let mut option = IterOption::default();
+        if let Some(end) = end_key {
+            option.set_upper_bound(end.encoded().clone());
```

Comment:
ditto

Reply:
I have updated Cargo.toml and server/service/kv.rs as well as certain code in storage/mod.rs. But this key.raw().unwrap() I have doubt if raw can be used for rawkv. Could you please double check this, thanks.

Reply:
You are right. It should be `encoded()`. I made a mistake.